components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java

```diff
@@ -140,6 +139,37 @@ public Tenant getTenant(String tenantUniqueID) throws TenantMgtException {
         return tenant;
     }
 
+    public Tenant getTenantByDomain(String domain) throws TenantMgtException {
+
+        TenantManager tenantManager = TenantMgtServiceComponent.getTenantManager();
+        Tenant tenant;
+        try {
+            int tenantID = tenantManager.getTenantId(domain);
+            tenant = (Tenant) tenantManager.getTenant(tenantID);
+            if (tenant == null) {
+                throw new TenantManagementClientException(ERROR_CODE_RESOURCE_NOT_FOUND.getCode(),
+                        String.format(ERROR_CODE_RESOURCE_NOT_FOUND.getMessage(), domain));
+            }
+        } catch (org.wso2.carbon.user.api.UserStoreException e) {
+            throw new TenantManagementServerException("Error while getting the tenant.", e);
+        }
+        return tenant;
+    }
+
+    public boolean isDomainAvailable(String domain) throws TenantMgtException {
+
+        TenantManager tenantManager = TenantMgtServiceComponent.getTenantManager();
+        try {
+            int tenantID = tenantManager.getTenantId(domain);
+            if (tenantID == -1) {
```

Comment:
Can you put a comment here saying -1 means, it is an invalid domain. 
So, there is no tenant domain registered with the given name. Thus the domain is available.

Reply:
Addressed.